includes/components/class-attribute.php

```diff
@@ -1860,4 +1863,25 @@ public function disable_jetpack_search( $enabled, $query ) {
 
 		return $enabled;
 	}
+
+	/**
+	 * Add archive page redirect.
+	 */
+	public function add_archive_redirect() {
+		foreach ( $this->get_models() as $model ) {
+			if ( is_post_type_archive( hp\prefix( $model ) ) ) {
```

Comment:
this functiona accepts an array of post types, the loop is not needed, you can get the current post type if it passes the check with get_post_type later